### ISSUE TITLE
As a passive LP, I want an easy way to add liquidity to Amarok AMMs by providing a single asset

### ISSUE THREAD

**Issue by alexwhte** [connext/nxtp#1472]
**Problem**
Users don't want to have to bridge/swap/pool to obtain the right amount of balanced pairs and LP 
This should be able to be achieved with just providing adopted on the desired chain.  

**Ideas to solve this**
User decides to LP USDC on Optimism. 
User enters amount 1000 USDC.
We show user estimated slippage and amounts. 
User clicks "Supply" in Pool UI
We call into automatically swap ~half of the provided funds into local nextUSDC, and then also add the funds into the pool with a single click. 

**Acceptance Criteria**
[ ] User choose to can add USDC for a given chain in the UI
[ ] Offchain and helper contracts automatically swaps ~50% and fund the AMM
[ ] This is visible and understandable to the users

**Implementation Steps**

**Other**

**Comment by alexwhte** [comment]
What will this look like with diamond pattern? 
- Create
- Test
- UI

**Comment by just-a-node** [comment]
Zapping into position isn't necessary for our pools because users can add liquidity in any proportion of tokenX/tokenY, even fully single-sided. Depending on the current reserves, the deposit can have positive or negative price impact on the pool. 

It would be difficult to determine the 'optimal' swap + deposit amounts because the state of the pool can change between the swap and the deposit, causing the first calculation for the swap to be obsolete.

Withdrawing from the pool will give back a split of X and Y tokens, not just one, unless the user specifies to withdraw in a single token (which will incur higher price impact).

In summary: 
- Update pool UI to allow for any amount of X,Y to be deposited (currently it is calculating amounts using an incorrect `rate` based on `virtualPrice` - this should be removed)
- Show user the price impact of their deposit/withdraw (pending #2151)
